AI Agents - initial docs (#2100)
port-labs/port-docs: docs/ai-agents/build-an-ai-agent.md

---
sidebar_position: 2
title: Build an AI agent
---

# Build an AI agent

import ClosedBetaFeatureNotice from '../generalTemplates/_closed_beta_feature_notice.md'
import Tabs from "@theme/Tabs"
import TabItem from "@theme/TabItem"

> <ClosedBetaFeatureNotice 

Author: @hadar-co actually I created this component, much like we have the _betaNotice component.
The reason:
1. I show it in three different pages.
2. Similar to beta, we will show in other places as well (like scorecards). I include the input of "contactMethod" since with closed beta, you can optionally show customers HOW to request access.

> <ClosedBetaFeatureNotice 

Reviewer: ok i see, using an external component makes sense.
Inside this component, let's use a standard callout instead of the custom one.

> <ClosedBetaFeatureNotice 

Author: Pushed an update

  accessRequestMethod={
    <span>please reach out to us by filling <a href='https://forms.gle/krhMY7c9JM8MyJJf7'>this form</a>.</span>
  } 
/>

## Getting started

Building an AI agent in Port involves several key steps. Let's walk through the process of creating an agent that can assist your developers with finding information and completing tasks.

<!-- IMAGE SUGGESTION: Add a screenshot of the AI agent creation page or dashboard here -->

## How to add a new AI agent

You can create a new AI agent in two ways:

> 1. Creating a new entity through the Port interface.

Author: For this version I removed the action option and clarified with screenshots how to add AI agent

3. Decide whether the agent can run the action automatically or requires approval.

<!-- IMAGE SUGGESTION: Add a screenshot of the action configuration interface -->

<Tabs groupId="action-approval" queryString>
<TabItem value="manual-approval" label="Manual Approval">

When configured for manual approval, the agent will:
1. Create a draft action based on the user's request.
2. Provide a link to the draft.
3. Allow the user to review and modify properties before execution.

<img src='/img/ai-agents/manual-approval-flow.png' width='80%' border='1px' />

This approach provides an additional safety layer, ensuring users can verify all parameters before execution.

</TabItem>
<TabItem value="automatic-execution" label="Automatic Execution">

When configured for automatic execution, the agent will:
1. Determine the appropriate parameters based on the user's request.
2. Execute the action immediately.
3. Provide a link to the action run.

<img src='/img/ai-agents/automatic-execution-flow.png' width='80%' border='1px' />

This approach streamlines workflows but should be used carefully, especially for actions with significant impact.

</TabItem>
</Tabs>

### Step 4: Define the prompt

> The prompt is your main tool for influencing your agent's success. You'll likely iterate on it as you observe how the agent responds to different queries. From our experience, a good prompt includes:
> 

Reviewer: ```suggestion
The prompt is your main tool for influencing your agent's success. You will likely iterate on it as you observe how the agent responds to different queries.  
In our experience, a good prompt includes:

```

2. **Domain context**: Share specifics about your data and terminology. For example, explain what "service" means in your organization's context.

3. **Response style**: Define how you want the agent to communicate - formal, friendly, concise, etc.

We recommend starting with a simple prompt and refining it based on how the agent performs. Here's a basic template:

```markdown 
You are an AI assistant for [Company Name]'s development team.

Your goal is to help developers [primary purpose, e.g., "find information about our services and assist with deployment processes"].

When answering questions:
- Be concise and direct
- Include links to relevant resources when available
- If you're unsure about something, acknowledge it rather than guessing

Key terminology:
- Service: [your definition]
- [Other important terms]: [definitions]
```

<!-- IMAGE SUGGESTION: Add a screenshot of the prompt configuration interface -->

### Step 5: Activate your agent

When you feel your agent is ready:

1. Set its status to "Active".
2. Start interacting with it through the [available interfaces](/ai-agents/interact-with-the-ai-agent).

## Examples

### Example 1: Service Information Agent

> This agent helps developers find information about services in your organization:

Reviewer: ```suggestion
This agent helps developers find information about services in their organization:
```
